torch/onnx/_internal/exporter.py

```diff
@@ -163,21 +164,194 @@ def serialize(
         destination.write(export_output.model_proto.SerializeToString())
 
 
+# TODO(bowbao): Add diagnostics for IO formatters.
+class InputFormatStep(abc.ABC):
```

Comment:
Have you considered making it a protocol?

Reply:
That's a good idea.